sdk/data/aztables/service_client.go

```diff
@@ -189,87 +173,109 @@ type ListTablesPage struct {
+	var value []*TableProperties
 
 	for _, v := range g.Value {
 		value = append(value, fromGeneratedTableResponseProperties(v))
 	}
 
-	return &ListTablesPage{
+	return ListTablesPage{
 		RawResponse:               g.RawResponse,
 		ContinuationNextTableName: g.XMSContinuationNextTableName,
 		ODataMetadata:             g.ODataMetadata,
 		Tables:                    value,
 	}
 }
 
-// ResponseProperties contains the properties for a single Table
-type ResponseProperties struct {
+// TableProperties contains the properties for a single Table
+type TableProperties struct {
 	// The edit link of the table.
 	ODataEditLink *string `json:"odata.editLink,omitempty"`
 
-	// The id of the table.
+	// The ID of the table.
 	ODataID *string `json:"odata.id,omitempty"`
 
 	// The odata type of the table.
 	ODataType *string `json:"odata.type,omitempty"`
 
 	// The name of the table.
-	TableName *string `json:"TableName,omitempty"`
+	Name *string `json:"TableName,omitempty"`
 }
 
 // Convets a generated TableResponseProperties to a ResponseProperties
-func fromGeneratedTableResponseProperties(g *generated.TableResponseProperties) *ResponseProperties {
+func fromGeneratedTableResponseProperties(g *generated.TableResponseProperties) *TableProperties {
 	if g == nil {
 		return nil
 	}
 
-	return &ResponseProperties{
-		TableName:     g.TableName,
+	return &TableProperties{
+		Name:          g.TableName,
 		ODataEditLink: g.ODataEditLink,
 		ODataID:       g.ODataID,
 		ODataType:     g.ODataType,
 	}
 }
 
-type tableQueryResponsePager struct {
+// ListTablesPager is a Pager for Table List operations
+//
+// Call NextPage first to fetch the next available page of results from the service.
+// If the fetched page contains results, the return value is true, else false.
+// Results fetched from the service can be evaluated by calling PageResponse on the Pager.
+// If the result is false, the value of Err() will indicate if an error occurred.
+//
+// PageResponse returns the results from the page most recently fetched from the service.
+type ListTablesPager struct {
 	client            *generated.TableClient
-	current           *generated.TableClientQueryResponse
+	current           generated.TableClientQueryResponse
 	tableQueryOptions *generated.TableClientQueryOptions
 	listOptions       *ListTablesOptions
-	err               error
+	nextTableName     *string
 }
 
 // NextPage fetches the next available page of results from the service.
 // If the fetched page contains results, the return value is true, else false.
 // Results fetched from the service can be evaulated by calling PageResponse on this Pager.
-func (p *tableQueryResponsePager) NextPage(ctx context.Context) bool {
-	if p.err != nil || (p.current != nil && p.current.XMSContinuationNextTableName == nil) {
-		return false
+func (p *ListTablesPager) NextPage(ctx context.Context) (ListTablesPage, error) {
```

Comment:
I think the return type should be named `ListTablesPageResponse` like our other response types.

Reply:
Fixed